es-app/src/guis/GuiSlideshowScreensaverOptions.cpp

```diff
@@ -0,0 +1,122 @@
+#include "guis/GuiSlideshowScreensaverOptions.h"
+#include "Window.h"
+#include "Settings.h"
+#include "views/ViewController.h"
+
+#include "components/SwitchComponent.h"
+#include "components/SliderComponent.h"
+#include "components/TextComponent.h"
+#include "components/OptionListComponent.h"
+#include "components/MenuComponent.h"
+#include "guis/GuiMsgBox.h"
+#include "guis/GuiTextEditPopup.h"
+#include "PowerSaver.h"
+
+GuiSlideshowScreensaverOptions::GuiSlideshowScreensaverOptions(Window* window, const char* title) : GuiScreensaverOptions(window, title)
+{
+	ComponentListRow row;
+
+	// image duration (seconds)
+	auto sss_image_sec = std::make_shared<SliderComponent>(mWindow, 3.f, 45.f, 1.f, "s");
+	sss_image_sec->setValue((float)(Settings::getInstance()->getInt("ScreenSaverSwapImageTimeout") / (1000)));
+	addWithLabel(row, "SWAP IMAGE AFTER (SECS)", sss_image_sec);
+	addSaveFunc([sss_image_sec] {
+		int playNextTimeout = (int)round(sss_image_sec->getValue()) * (1000);
+		Settings::getInstance()->setInt("ScreenSaverSwapImageTimeout", playNextTimeout);
+		PowerSaver::updateTimeouts();
+	});
+
+	// stretch
+	auto sss_stretch = std::make_shared<SwitchComponent>(mWindow);
+	sss_stretch->setState(Settings::getInstance()->getBool("SlideshowScreenSaverStretch"));
+	addWithLabel(row, "STRETCH IMAGES", sss_stretch);
+	addSaveFunc([sss_stretch] {
+		Settings::getInstance()->setBool("SlideshowScreenSaverStretch", sss_stretch->getState());
+	});
+
+	// background audio file
+	auto sss_bg_audio_file = std::make_shared<TextComponent>(mWindow);
+	addEditableTextComponent(row, "BACKGROUND AUDIO", sss_bg_audio_file, Settings::getInstance()->getString("SlideshowScreenSaverBackgroundAudioFile"));
+	addSaveFunc([sss_bg_audio_file] {
+		Settings::getInstance()->setString("SlideshowScreenSaverBackgroundAudioFile", sss_bg_audio_file->getValue());
+	});
+
+	// image source
+	auto sss_source = std::make_shared< OptionListComponent<std::string> >(mWindow, "IMAGE SOURCE", false);
+	std::vector<std::string> image_sources;
+	image_sources.push_back("game list");
+	image_sources.push_back("custom");
+	for(auto it = image_sources.begin(); it != image_sources.end(); it++)
```

Comment:
Considering there will always be two options, consider using a switch_component? Can be labelled "Use custom source" or anything else

Reply:
I was thinking about a 3rd option to pull an image from a URL.  I haven't tried implementing it yet, but I could envision having a webserver return the image to display.  Each time the screensaver wants to cycle the image, it would request a new image from the webserver.  The webserver could be as simple as a php script that decides which image to serve.  I was thinking that I would run the webserver on my LAN (primarily to not waste bandwidth), but in theory the webserver could be hosted anywhere.  Does that sound like a reasonable feature?

Reply:
I think that should be done in a separate PR if needed. Lets not mix features